src/Network/Network/VirtualNetworkGateway/StartAzVirtualNetworkGatewayPacketCaptureCommand.cs

```diff
@@ -0,0 +1,119 @@
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Network.Models;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Network;
+using Microsoft.Azure.Management.Network.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Network.VirtualNetworkGateway
+{
+    [Cmdlet("Start", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "VirtualnetworkGatewayPacketCapture", 
+        DefaultParameterSetName = "ByName", SupportsShouldProcess = true), OutputType(typeof(PSVirtualNetworkGatewayPacketCaptureResult))]
+    public class StartAzVirtualNetworkGatewayPacketCaptureCommand : VirtualNetworkGatewayBaseCmdlet
+    {
+        [Parameter(
+            ParameterSetName = "ByName",
+            Mandatory = true,
+            HelpMessage = "The resource group name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Alias("ResourceName", "VirtualNetworkGatewayName", "GatewayName")]
+        [Parameter(
+            ParameterSetName = "ByName",
+            Mandatory = true,
+            HelpMessage = "The virtual network gateway name where packet capture is to be started.")]
+        [ResourceNameCompleter("Microsoft.Network/virtualNetworkGateways", "ResourceGroupName")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("VirtualNetworkGateway")]
+        [Parameter(
+            ParameterSetName = "ByInputObject",
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = "The virtual network gateway object where packet capture to be started.")]
+        [ValidateNotNullOrEmpty]
+        public PSVirtualNetworkGateway InputObject { get; set; }
+
+        [Parameter(
+            ParameterSetName = "ByResourceId",
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "The Azure resource ID of the VirtualNetworkGateway where packet capture to be started.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Parameter(
+            Mandatory = false,
+            HelpMessage = "Filter options for start packet capture on virtual network gateway.")]
+        public string FilterData { get; set; }
```

Comment:
What can the user enter here?  Is there any way we can guide them toward appropriate (functional) filter expressions?

Reply:
We are using this field to provide a mechanism to users to provide filters queries to have a functionality similar to what WireShark or other packet capture tools provide. This can be regEx, OData or JSON in the future. For now we are only supporting JSON. We will be proving public documentation on how these will be defined and we will also be adding helper  powershell commands in the future to generate these quires.